### PR TITLE
Adding Form Validation to the DateInput

### PR DESCRIPTION
**Pull request by lgarceau768**
Fix to #499 Row 87
Adds validation to prompt the user with an error whenever they input a non valid character into the DateInput field

### CODE REVIEW

**Review by fpigeonjr — APPROVED** [review]
Your demo looked good. 
No letters allowed. Only numbers.

File: services/dateHelper.js [hunk 1]
```diff
@@ -117,6 +117,14 @@ function validInputs(month, day, year) {
 }
 
 function checkDateValid(month, day, year) {
+  const invalidDateInput = [month, day, year].some((element) => {
+    if (element !== "") {
+      return !/\d+$/.test(element)
+    } else return false
+  })
```
Comment (wesley-dean-gsa) [inline services/dateHelper.js:124]: I'm sorry, my lack of JavaScript knowledge is inhibiting my ability to understand this.

**Comment by r-bartlett-gsa** [comment]
@lgarceau768 / @fpigeonjr Any date in 2022 returns as invalid: https://www.screencast.com/t/lwovgnOi9Z9d If 12/31/2021 is entered then it's all good. I assume it's checking against current year vs current date? Can you please fix that.

File: services/dateHelper.js (hunk 1 above)
Comment (lgarceau768) [reply services/dateHelper.js:124]: It is using a Regex expression to loop through the array and test for when an input has something other then numbers and stops looping when there is an input other than numbers but not the default empty input

**Comment by r-bartlett-gsa** [comment]
@lgarceau768 Is there a way to validate against the current date (month, date and year), not just against the current year?

**Comment by lgarceau768** [comment]
> @lgarceau768 / @fpigeonjr Any date in 2022 returns as invalid: https://www.screencast.com/t/lwovgnOi9Z9d If 12/31/2021 is entered then it's all good. I assume it's checking against current year vs current date? Can you please fix that.

This should be resolved now!

**Pull request merged**